docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+
+### Local and External Kits
+
+During the build process, there will be a distinction between *local kits* and *external kits*.
+A *local kit* is any kit where the packages are defined in the project tree and the maintainer wants to build them from source as part of the Bottlerocket build.
+An *external kit* is a kit that is used by pulling it from a docker registry, i.e. the container image has previously been built.
+
+It should not be a requirement for a maintainer to push a kit to a container registry.
+In other words, it should not be necessary to publish a kit in order to create a Bottlerocket variant.
+This is where *local kits* come in.
+Kits defined in the project tree need not be pre-built before a `twoliter build` command can succeed.
+Twoliter will understand that the local kits need to be built before the rest of the variant build can proceed.
+
+### Kits Project Directory
+
+Currently, there are three directories used for sourcecode when building Bottlerocket:
+- `sources`: A Cargo workspace for first-party Rust code.
+- `packages`: A collection of Rust packages that build RPM packages.
+- `variants`: A Cargo workspace for defining what Bottlerocket variants.
+
+A new directory will be added: `kits`.
+This will be a Cargo workspace with one Cargo package for each kit.
+
+```
+kits
+├── aws
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── core
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── ecs
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── k8s
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── metal
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+└── vmware
+    ├── build.rs
+    ├── Cargo.toml
+    └── lib.rs
+```
+
+A variant such as `metal-k8s-1.22` would consume the necessary kits.
+For example, it would reference `core-kit`, `k8s-kit` and `metal-kit`.
+To add `iam-authenticator`, `aws-kit` would be added.
+
+It may be convenient for certain build steps to have a single package representing a kit, such as `bottlerocket-aarch64-core-kit-0.1-1.aarch64.rpm`
+If so, these spec files could be added to the above directory tree.
+It is also possible to generate these spec files as they would be simple aggregations of all the packages in the kit.
+
+### Kit Structure
+
+A kit container will be built using `FROM scratch` because it does not need a runtime environment.
+It's directory structure will look like this:
+
+```sh
+# tree /local --dirsfirst
+/local
+├── etc
+│   └── yum.repos.d
+│       └── bottlerocket-core-kit.repo
+└── kits
+    └── bottlerocket-core-kit
+        ├── archives
+        │   └── bottlerocket-x86_64-kernel-5.15-archive-5.15.102-1.x86_64.rpm
+        ├── bootconfig
+        │   ├── empty-bootconfig.data
+        │   └── qemu-x86-console-bootconfig.data
+        ├── licenses
+        │   ├── COPYRIGHT
+        │   ├── LICENSE-APACHE
+        │   └── LICENSE-MIT
+        ├── migrations
+        │   └── bottlerocket-x86_64-migrations-0.0-0.x86_64.rpm
```

Comment:
Can we strip out all the non-RPM and non-repo content here?

This stuff shouldn't be published along with the kit.

Reply:
> Can we strip out all the non-RPM and non-repo content here?

Note: this is about the licenses.

Reply:
Note: remove migrations and other tech debt discussions from the doc.